hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/HivePartitionUtil.java

```diff
@@ -48,4 +49,16 @@ public static String getPartitionClauseForDrop(String partition, PartitionValueE
     }
     return String.join("/", partBuilder);
   }
+
+  public static Boolean partitionExists(IMetaStoreClient client, String tableName, String partitionPath,
+                                        PartitionValueExtractor partitionValueExtractor, HiveSyncConfig config) {
+    Partition newPartition;
+    try {
+      List<String> partitionValues = partitionValueExtractor.extractPartitionValuesInPath(partitionPath);
+      newPartition = client.getPartition(config.databaseName, tableName, partitionValues);
+    } catch (Exception ignored) {
+      newPartition = null;
+    }
```

Comment:
<img width="717" alt="Screen Shot 2022-03-29 at 11 14 27 PM" src="https://user-images.githubusercontent.com/2701446/160645043-1e350985-b655-4ecb-8fe9-1fb6f2c8e892.png">

we shouldn't be catching all exceptions; it could mask other issues. Instead, we only catch NoSuchObjectException for partition non-exist case. Other exceptions, we should re-throw a hoodie exception containing it.